src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs

```diff
@@ -11,6 +11,9 @@ namespace System.Runtime.CompilerServices
 {
     public static partial class RuntimeHelpers
     {
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        public static extern void MarkDontNeedReadOnlyPages();
```

Comment:
Public APIs needs API proposal in CoreFX: https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md

Reply:
I think it would be better to think about this API as "flush caches" - not limit it to R/O pages.